tools/clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1727,7 +1727,7 @@ static void ParseLangArgs(LangOptions &Opts, ArgList &Args, InputKind IK,
   else {
     try {
       Opts.HLSLVersion = std::stoi(std::string(ver));
-      if (Opts.HLSLVersion < 2015 || Opts.HLSLVersion > 2018) {
+      if (Opts.HLSLVersion < 2015 || Opts.HLSLVersion > 2021) {
```

Comment:
Theoretically, aren't 2019 and 2020 also errors?

Reply:
Sadly, we just range check this value... We totally should make this an enum rather than an integer and do a real conversion. I've got that on my mental to do list.